apps/qubit/modules/object/actions/exportAction.class.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class ObjectExportAction extends DefaultEditAction
+{
+  // Arrays not allowed in class constants
+  public static
+    $NAMES = array(
+      'levels');
+
+  private $choices = array();
+
+  protected function earlyExecute()
+  {
+    $this->form->getValidatorSchema()->setOption('allow_extra_fields', true);
+  }
+
+  protected function addField($name)
+  {
+    switch ($name)
+    {
+      case 'levels':
+        $this->form->setValidator('levels', new sfValidatorPass);
+
+        foreach (QubitTerm::getLevelsOfDescription() as $item)
+        {
+          $this->choices[$item->id] = $item->__toString();
+        }
+
+        $size = count($this->choices);
+        if ($size === 0)
+        {
+          $size = 4;
+        }
+
+        $this->form->setWidget('levels', new sfWidgetFormSelect(array('choices' => $this->choices, 'multiple' => true), array('size' => $size)));
+
+        break;
+
+      default:
+        return parent::addField($name);
+    }
+  }
+
+  protected function processField($field)
+  {
+    switch ($field->getName())
+    {
+      case 'levels':
+        $this->levels = $this->form->getValue('levels');
+        if (empty($this->levels))
+        {
+          $this->levels = array();
+        }
+
+        break;
+    }
+  }
+
+  protected function doBackgroundExport($request)
+  {
+    // Export type, CSV or XML?
+    $exportType = $request->getParameter('format', 'xml');
+
+    // We will use this later to redirect users back to the export page
+    if (isset($this->getRoute()->resource))
+    {
+      $exportRoute = array($this->getRoute()->resource, 'module' => 'object', 'action' => 'export', 'type' => $exportType);
+    }
+    else
+    {
+      $exportRoute = array('module' => 'object', 'action' => 'export', 'type' => $exportType);
+    }
```

Comment:
Actually, this looks like it comes from the import page, where you can import directly to a resource, but it should not be needed in the export page (AFAIK it can't be reached from a resource). And, as I said in other comments, if you move the `$this->limit` population outside the method check in the execute function, you should be able to access `$type` in the template in all cases.